### PR TITLE
feat: Test cases for Scheduled Message feature

### PR DESCRIPTION
**Pull request by yasserfaraazkhan**
JIRA: https://mattermost.atlassian.net/browse/MM-60375

### CODE REVIEW

File: data/test-cases/channels/schedule-messages/1-Create-Schedule-Messages.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+**Step 1**
+
+### Scenario 1: Able to schedule a message to a future Date in GM/Channel
+
+Steps:
+
+1. Log into User's account.
+2. Type a message in the GM/Channel post text box.
+3. Click on the Dropdown beside Send button.
+4. Choose a pre-defined Date from the options.
```
Comment (marianunez) [inline data/test-cases/channels/schedule-messages/1-Create-Schedule-Messages.md:45]: Can we verify the timezone behavior for scheduling? I believe the expected behavior is the date and time dialog is in the users configured timezone, can you confirm @harshilsharma63 ?

File: data/test-cases/channels/schedule-messages/1-Create-Schedule-Messages.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
+---
+# (Required) Ensure all values are filled up
+name: "Verify User can create a scheduled message in DM/GM/Channel"
```
Comment (marianunez) [inline data/test-cases/channels/schedule-messages/1-Create-Schedule-Messages.md:3]: Can we also test messages that include attachments and post priority set?

**Review by marianunez — CHANGES_REQUESTED** [review]
Thanks @yasserfaraazkhan! Added some comments on additional cases we want to include. 

Also, for the first version I believe we will be supporting moving a draft to a scheduled post. Is this correct @harshilsharma63 ? If so, we need to include test cases for add/remove schedule to drafts

File: data/test-cases/channels/schedule-messages/1-Create-Schedule-Messages.md (hunk 2 above)
Comment (yasserfaraazkhan) [reply data/test-cases/channels/schedule-messages/1-Create-Schedule-Messages.md:3]: Sure Maria. Had made a checklist. Will also add in zephyr.

File: data/test-cases/channels/schedule-messages/1-Create-Schedule-Messages.md (hunk 1 above)
Comment (harshilsharma63) [reply data/test-cases/channels/schedule-messages/1-Create-Schedule-Messages.md:45]: Thats correct, it should be in the user's configured TZ, not the timezone of the browser.

File: data/test-cases/channels/schedule-messages/4-Failed-Schedule-Message.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+1. Log into User A's account and create a Root post in a channel.
+2. Reply to the root post with some messages.
+3. Log into User B's account and create a threaded schedule message to the root post.
+4. From User A's account, Using **Move Threads** feature, move the root post to a private Channel where User B is not a member.
+5. Log into User B's account.
+6. Goto Scheduled messages tab.
+7. Verify the message has a tag `Not in the channel` in red alert.
+8. Verify the User can see `No Channel` in the Channels section.
+
+### Scenario 7: Verify Schedule message fails if the root post is moved to another team channel.
```
Comment (lindalumitchell) [inline data/test-cases/channels/schedule-messages/4-Failed-Schedule-Message.md:102]:
```suggestion
### Scenario 7: Verify Schedule message fails if the root post is moved to another team.
```

File: data/test-cases/channels/schedule-messages/4-Failed-Schedule-Message.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+1. Log into User A's account and create a Root post in a channel.
+2. Reply to the root post with some messages.
+3. Log into User B's account and create a threaded schedule message to the root post.
```
Comment (lindalumitchell) [inline data/test-cases/channels/schedule-messages/4-Failed-Schedule-Message.md:106]:
```suggestion
3. Log into User B's account and create a threaded scheduled message on the root post.
```

File: data/test-cases/channels/schedule-messages/4-Failed-Schedule-Message.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+4. From User A's account, Using **Move Threads** feature, move the root post to another Team where User B is not a member.
+5. Log into User B's account.
+6. Goto Scheduled messages tab.
```
Comment (lindalumitchell) [inline data/test-cases/channels/schedule-messages/4-Failed-Schedule-Message.md:109]:
```suggestion
6. Go to Scheduled messages tab.
```

File: data/test-cases/channels/schedule-messages/5-Failed-Server-outage-Schedule-Message.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
+---
+# (Required) Ensure all values are filled up
+name: "Verify if the Failed Scheduled message is sent out after server recovered from and Outage issue"
```
Comment (lindalumitchell) [inline data/test-cases/channels/schedule-messages/5-Failed-Server-outage-Schedule-Message.md:3]:
```suggestion
name: "Verify if the Failed Scheduled message is sent out after server recovered from an outage issue"
```

File: data/test-cases/channels/schedule-messages/5-Failed-Server-outage-Schedule-Message.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+status: Active
+priority: Normal
+folder: Schedule Messages
+authors: "@yasserfaraazkhan"
+team_ownership:
+- ICU
+priority_p1_to_p4: P2 - Core Functions (Do core functions work?)
```
Comment (lindalumitchell) [inline data/test-cases/channels/schedule-messages/5-Failed-Server-outage-Schedule-Message.md:10]:
```suggestion
priority_p1_to_p4: P3 - Deep Functions (Do extensive scenarios work?)
```

File: data/test-cases/channels/schedule-messages/5-Failed-Server-outage-Schedule-Message.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+1. Should have scheduled messages in a DM/GM/Channel.
+
+### Scenario 1: The server recovers within a timestamp of 30 minutes and the message is sent out
+
+Steps:
+
+1. Log into User's account.
+2. Simulate server outage for a time period when the message was scheduled.
+   - Option 1: User Jmeter or k6 tool to generate huge traffic on the server.
```
Comment (lindalumitchell) [inline data/test-cases/channels/schedule-messages/5-Failed-Server-outage-Schedule-Message.md:48]:
```suggestion
   - Option 1: Use Jmeter or k6 tool to generate huge traffic on the server.
```

File: data/test-cases/channels/schedule-messages/5-Failed-Server-outage-Schedule-Message.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+   - Option 2: If the instance is created using Cloud plugin, try options like `--database` or `hibernate`
+3. Wait till the scheduled message fails due to server outage
+4. Once the serer recovers, Verify if the scheduled message is sent out within 30 mins.
+
+### Scenario 2: The server does not recover within a timestamp of 30 minutes. Messages is marked as Failed.
```
Comment (lindalumitchell) [inline data/test-cases/channels/schedule-messages/5-Failed-Server-outage-Schedule-Message.md:53]:
```suggestion
### Scenario 2: The server does not recover within a timestamp of 30 minutes. Message is marked as Failed.
```

File: data/test-cases/channels/schedule-messages/6-Add-Schedule-to-Draft.md [hunk 10]
```diff
@@ -0,0 +1,54 @@
+---
+# (Required) Ensure all values are filled up
+name: "Verify User can create a Schedule message from the Draft Page"
+status: Active
+priority: Normal
+folder: Schedule Messages
+authors: "@yasserfaraazkhan"
+team_ownership:
+- ICU
+priority_p1_to_p4: P2 - Core Functions (Do core functions work?)
```
Comment (lindalumitchell) [inline data/test-cases/channels/schedule-messages/6-Add-Schedule-to-Draft.md:10]:
```suggestion
priority_p1_to_p4: P3 - Deep Functions (Do extensive scenarios work?)
```

File: data/test-cases/channels/schedule-messages/6-Add-Schedule-to-Draft.md [hunk 11]
```diff
@@ -0,0 +1,54 @@
+---
+# (Required) Ensure all values are filled up
+name: "Verify User can create a Schedule message from the Draft Page"
```
Comment (lindalumitchell) [inline data/test-cases/channels/schedule-messages/6-Add-Schedule-to-Draft.md:3]:
```suggestion
name: "Verify User can create a scheduled message from the Draft Page"
```

File: data/test-cases/channels/schedule-messages/6-Add-Schedule-to-Draft.md [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+
+**Step 1**
+
+### Scenario 1: Able to add schedule to the Draft message.
+
+Steps:
+
+1. Log into User's account.
+2. Type a message in the DM/GM/Channel post text box to create a Draft message.
+3. Goto Drafts page.
```
Comment (lindalumitchell) [inline data/test-cases/channels/schedule-messages/6-Add-Schedule-to-Draft.md:44]:
```suggestion
3. Go to Drafts page.
```

File: data/test-cases/channels/schedule-messages/6-Add-Schedule-to-Draft.md [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+4. Verify the Draft is created and has channel name and date of creation.
+5. Hover on the Draft.
+6. Click on the Schedule icon.
+7. Verify a modal to select Date and time appears.
+8. Select a Future date and click on `Schedule` button.
+9. Verify the message is not seen under Drafts tab.
+10. Goto Scheduled tab.
+11. Verify the Scheduled message appears on the top.
+12. GOto the DM/GM/Channel where the draft was created.
```
Comment (lindalumitchell) [inline data/test-cases/channels/schedule-messages/6-Add-Schedule-to-Draft.md:53]:
```suggestion
12. Go to the DM/GM/Channel where the draft was created.
```

File: data/test-cases/channels/schedule-messages/7-Remove-Schedule-and-Move-to-Draft.md [hunk 14]
```diff
@@ -0,0 +1,53 @@
+---
+# (Required) Ensure all values are filled up
+name: "Verify User can remove scheduled setting from the Scheduled message"
+status: Active
+priority: Normal
+folder: Schedule Messages
+authors: "@yasserfaraazkhan"
+team_ownership:
+- ICU
+priority_p1_to_p4: P2 - Core Functions (Do core functions work?)
```
Comment (lindalumitchell) [inline data/test-cases/channels/schedule-messages/7-Remove-Schedule-and-Move-to-Draft.md:10]:
```suggestion
priority_p1_to_p4: P3 - Deep Functions (Do extensive scenarios work?)
```

File: data/test-cases/channels/schedule-messages/7-Remove-Schedule-and-Move-to-Draft.md [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+---
+
+**Step 1**
+
+### Scenario 1: Able to add schedule to the Draft message.
+
+Steps:
+
+1. Log into User's account.
+2. Goto Scheduled tab.
```
Comment (lindalumitchell) [inline data/test-cases/channels/schedule-messages/7-Remove-Schedule-and-Move-to-Draft.md:43]:
```suggestion
2. Go to Scheduled tab.
```

File: data/test-cases/channels/schedule-messages/7-Remove-Schedule-and-Move-to-Draft.md [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+3. Verify the Draft is created and has channel name and date of creation.
+4. Hover on the Draft.
+5. Click on the Schedule icon.
+6. Verify a modal to select Date and time appears.
+7. Select a Future date and click on `Schedule` button.
+8. Verify the message is not seen under Drafts tab.
+9. Goto Scheduled tab.
```
Comment (lindalumitchell) [inline data/test-cases/channels/schedule-messages/7-Remove-Schedule-and-Move-to-Draft.md:50]:
```suggestion
9. Go to Scheduled tab.
```

File: data/test-cases/channels/schedule-messages/7-Remove-Schedule-and-Move-to-Draft.md [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+10. Verify the Scheduled message appears on the top.
+11. GOto the DM/GM/Channel where the draft was created.
```
Comment (lindalumitchell) [inline data/test-cases/channels/schedule-messages/7-Remove-Schedule-and-Move-to-Draft.md:52]:
```suggestion
11. Go to the DM/GM/Channel where the draft was created.
```

**Review by lindalumitchell — APPROVED** [review]
Nice, @yasserfaraazkhan! I added a bunch of nitpicky suggestions, plus I've started auditing the priorities set on our test cases, so these should be P3 in our custom priority field, as noted. Approving since the substance looks great; you can decide about my suggestions. :)

**Review by lindalumitchell — APPROVED** [review]
Nice, @yasserfaraazkhan! I added a bunch of nitpicky suggestions, plus I've started auditing the priorities set on our test cases, so I marked that these should be P3 in our custom priority field. Approving since the substance looks great on the test cases; you can decide about my suggestions. :)

File: data/test-cases/channels/schedule-messages/1-Create-Schedule-Messages.md (hunk 1 above)
Comment (yasserfaraazkhan) [reply data/test-cases/channels/schedule-messages/1-Create-Schedule-Messages.md:45]: added

**Review by harshilsharma63 — APPROVED** [review]

**Review by marianunez — APPROVED** [review]
Thanks @yasserfaraazkhan!

**Pull request merged**